cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/image/marketplace/AzureImageTermsSignerService.java

```diff
@@ -69,18 +72,19 @@ public boolean isSigned(String subscriptionId, AzureMarketplaceImage azureMarket
      */
     @Retryable(maxAttempts = 15, backoff = @Backoff(delay = 1000, multiplier = 2, maxDelay = 10000))
     public void sign(String subscriptionId, AzureMarketplaceImage azureMarketplaceImage, AzureClient azureClient) {
+        URI agreementUri = getAgreementUri(subscriptionId, azureMarketplaceImage);
         URI signUri = getSignUri(subscriptionId, azureMarketplaceImage);
         ErrorMessageBuilder errorMessageBuilder =
                 new ErrorMessageBuilder(String.format(SIGN_ERROR_MESSAGE_TEMPLATE, azureMarketplaceImage)).withPostfix(SIGN_ERROR_MESSAGE_HINTS);
         String token = getToken(azureClient, errorMessageBuilder);
 
         try {
-            AzureImageTerms azureImageTerms = azureRestOperationsService.httpGet(signUri, AzureImageTerms.class, token);
+            AzureImageTerms azureImageTerms = azureRestOperationsService.httpGet(agreementUri, AzureImageTerms.class, token);
             LOGGER.debug("Image terms and conditions received for image {} is : {}", azureMarketplaceImage, azureImageTerms);
 
             azureImageTerms.getProperties().setAccepted(true);
 
-            AzureImageTerms responseImageTerms = azureRestOperationsService.httpPut(signUri, azureImageTerms, AzureImageTerms.class, token);
+            AzureImageTerms responseImageTerms = azureRestOperationsService.httpPost(signUri, null, AzureImageTerms.class, token);
```

Comment:
I tried locally and for me the POST endpoint with the above URL gives an error:
`Bad Request({"error":{"code":"BadRequest","message":"You must accept the legal terms through the Azure portal (http://go.microsoft.com/fwlink/?LinkId=534873). Correlation Id: '1bdf2fe2-542a-47ce-9a14-80abe382499d'."}})`

I also remember that I had to do it over the PUT endpoint. I quickly tested it via the command `az rest` and agreement accept and decline did work.